### DISCUSSION TITLE
How to ignore spaces between a class member and its type in "no-multi-spaces"

### DISCUSSION THREAD

**Discussion by jhoryong** [eslint/eslint#17607] [Q&A] [answered]
I want to organize my class and its members like below for readability:
```ts
class Exmple {
  private a:        string;
// or
  private b         : number;
}
```
but the `"no-multi-spaces"` rule disallows spaces between a class member variable and its type declaration.
I tried:
```
/* eslint no-multi-spaces: [{ exceptions: { "AssignmentExpression": true, "VariableDeclarator": true } }] */
```
rule does not fix my problem.

Is there an option to ignore typing in class member declarations? Or are there any workarounds?

My environment:
package.json:
```
    "eslint": "^8.30.0",
    "eslint-config-airbnb-base": "^15.0.0",
    "eslint-import-resolver-typescript": "^2.7.1",
    "eslint-plugin-check-file": "^1.3.0",
    "eslint-plugin-import": "^2.26.0",
    "eslint-webpack-plugin": "^3.1.1",
```
.eslintrc
```
"plugins": [
        "check-file",
        "@typescript-eslint",
        "import"
    ],
    "extends": [
        "airbnb-base",
        "plugin:@typescript-eslint/eslint-recommended"
    ],
```

**Comment by mdjermanovic** [comment, marked as answer]
> ```ts
> private a:        string;
> ```

This would be `TSTypeAnnotation` exception.

> ```ts
> private b         : number;
> ```

And this would be `PropertyDefinition` exception.

Here's typescript-eslint playground demo:

https://typescript-eslint.io/play/#ts=5.2.2&fileType=.tsx&code=LAKA9AVABApgzgGwJYDsAuUUHsC0BbAVwTSRzgAcBDAY3gC4oBtAIhgCc2s3mAaKAb1gAPWuRJYUcBoNBQozACoBlBQE9yMAIIpsaSuJTMGaNgRg9Z8gAqcNbNKoAiMAGaokBo1BNnQAXz8AXSgIMFBQagRKODgoAFEhPHIEGAFLcjYkADd9VMoGOUK5OBNUAHMAbkswMCgudMyctFSAIyL2hhQCPBb2KpA-IA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

**Reply by jhoryong** [reply]
Wow that works like a charm. Thanks a bunch!

One thing, I want class field declaration to also follow the same style. For example:
```ts
class Example {
  private a:        string;
// or
  private b         : number;
// class field declaration
  private c         = 1;
}
```
I get "no-multi-spaces" warning between `c` and `=`. Is there any solution to this?



Also, when I tried `PropertyDefinition ` method, it collides with `@typescript-eslint/type-annotation-spacing`, which is set to `error`. 
The error states: `Unexpected space between the '?' and the ':'` (or just`:` when there is no `?`)
If I were to use  `PropertyDefinition ` method, how would I need to change  `@typescript-eslint/type-annotation-spacing`?

**Comment by mdjermanovic** [comment]
> I get "no-multi-spaces" warning between `c` and `=`. Is there any solution to this?

`PropertyDefinition` exception also covers this. Here's the example with `private c         = 1`:

https://typescript-eslint.io/play/#ts=5.2.2&fileType=.tsx&code=LAKA9AVABApgzgGwJYDsAuUUHsC0BbAVwTSRzgAcBDAY3gC4oBtAIhgCc2s3mAaKAb1gAPWuRJYUcBoNBQozACoBlBQE9yMAIIpsaSuJTMGaNgRg9Z8gAqcNbNKoAiMAGaokBo1BNnQAXz8AXSgIMFBQagRKODgoAFEhPHIEGAFLcjYkADd9VMoGOUK5OBNUAHMAbkswMCgudMyctFSAIyL2hhQCPBb2KpA5DOzcqGp2ooBeKABGfr8gA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false


> Also, when I tried `PropertyDefinition ` method, it collides with `@typescript-eslint/type-annotation-spacing`, which is set to `error`.
> The error states: `Unexpected space between the '?' and the ':'` (or just`:` when there is no `?`)
> If I were to use `PropertyDefinition ` method, how would I need to change `@typescript-eslint/type-annotation-spacing`?

For configuring typescript-eslint plugin rules, it's better to ask in the the typescript-eslint repo:

https://github.com/typescript-eslint/typescript-eslint
